tests/Responses/ajax/product/index_response.json

```diff
@@ -0,0 +1,160 @@
+{
+    "page": 1,
+    "limit": 10,
+    "pages": 3,
+    "total": 22,
+    "_links": {
+        "self": {
+            "href": "\/admin\/ajax\/products\/?page=1&limit=10"
+        },
+        "first": {
+            "href": "\/admin\/ajax\/products\/?page=1&limit=10"
+        },
+        "last": {
+            "href": "\/admin\/ajax\/products\/?page=3&limit=10"
+        },
+        "next": {
+            "href": "\/admin\/ajax\/products\/?page=2&limit=10"
+        }
+    },
+    "_embedded": {
+        "items": [
+            {
+                "id": @integer@,
+                "name": "Breaking bad mug",
+                "code": "MUG_BB",
+                "options": [],
+                "averageRating": 0,
+                "images": [],
+                "_links": {
+                    "self": {
+                        "href": "\/api\/v1\/products\/MUG_BB"
+                    }
+                }
```

Comment:
If we are allowing for a BC break here, I would go further, and reduce amount of data to the bare minimum. (perhaps only name and code?)

Reply:
To do this, the `serialization_groups` will have to be changed to a new one. So all Ajax requests should get a new `AjaxRequest` `validation_groups` to only retrieve the strict minimum info. Another PR should be done to target this thing I suppose.

Reply:
I have to retest with the `serialization_group` named `Autocomplete`, I already tested it a while ago and there was more problem using it, I will see if I can take care of it.

Reply:
If I try to change the `serialization_groups ` to `Autocomplete ` then the json result is `{}` because the `Hateoas\Representation\PaginatedRepresentation` owning the data is serializable only with the serialization group `Default`. Other ajax routes override the repository method so the data retrieved is a simple array and not a `PaginatedRepresentation`.

So if I add a repository method like `findAll` to the route definition it will returns a simple json `array` of `Product` without other info like `total`, `pages`, etc. And then the javascript [use to display those results](https://github.com/Sylius/Sylius/tree/master/src/Sylius/Bundle/UiBundle/Resources/private/js/sylius-product-auto-complete.js) have to be deleted in favor of the wider [`jQuery.autoComplete`](https://github.com/Sylius/Sylius/tree/master/src/Sylius/Bundle/UiBundle/Resources/private/js/sylius-auto-complete.js) function use with the other autocomplete, and maybe more than this change is needed. Less code but this will be a BC break.

So @lchrusciel, @pamil do I have to introduce this BC break ?

Reply:
I'd stay with the current solution then, WDYT @lchrusciel?